src/Demos/HSP.lagda

```diff
@@ -1,104 +1,81 @@
+proof and derive a contradiction. To resolve these issues, as well as general improvements,
+we have rewritten parts of the library and have
+developed a new proof of the HSP theorem. We are confident that the new
+proof---in the \agdaalgebras library, ver.~2.0.1~\cite{ualib_v2.0.1}---is
+constructive and correct.
+
+What follows is a self-contained \agda proof of the HSP theorem.  This is done by
+extracting into a single module, as a literate \agda document\footnote{See the file
+(src/Demoes/\HSPlagda) in the repository, using \agda ver.2.6.2 and \agdastdlib ver.1.7.},
+a subset of the \agdaalgebras library, including only
+the pieces needed for the proof.
 \ifshort
-In order to present the proof in at most 20 pages, we omit some of the formalities,
+For spaces reasons, we do elide some inessential parts,
 but strive to preserve the essential content and character of the development.
-We leave out the routine or overly technical components, as well as anything that does not
+More specifically,routine or overly technical components, as well as anything that does not
 seem to offer any insight into the central ideas of the proof.\footnote{The unabridged
-version of this paper,~\cite{DeMeo:2021}, presents the formal proof in its entirety.}
-%and, apart from a few dozen imports from the \agdastdlib, is self-contained.}
+version~\cite{DeMeo:2021} presents the full proof.}
 \else
-We include here every line of code of our formal proof of Birkhoff's theorem
+We include here every line of code of our proof of Birkhoff's theorem
 in a single \agda module.  Apart from a few dozen imports from the \agdastdlib, the module
-is self-contains.
+is self-contained.
 \fi
 
-In the course of this presentation we highlight some of the challenging aspects of
-formalizing the basic definitions and theorems of universal algebra in type theory.
-%In particular, we touch on some technical issues that arise when attempting
-%to constructively prove basic results of our field.  Nonetheless,
-One positive contribution of this project is that it lends support to the claim that
-dependent type theory and the \agda language, despite the technical demands they place on
-the user, are accessible to working mathematicians (such as ourselves) who possess
-sufficient patience and resolve to codify their work in type theory in order to formally verify
-their results with a proof assistant.
+We highlight some of the challenging aspects of formalizing universal algebra in type theory.
+On the positive side, the patient mathematician with enough resolve and the will to learn
+dependent type theory can reap the rewards of the further insights that mechanization provide.
+% One positive contribution of this project is that it lends support to the claim that
+% dependent type theory and the \agda language, despite the technical demands they place on
+% the user, are accessible to working mathematicians (such as ourselves) who possess
```

Comment:
Yes, this was overly wordy, but I would like to express the bit that's commented out now.  Maybe we can use it in a conclusion/future work section (which I think we are still missing).

Reply:
I think there's been quite a few papers that have reached a similar conclusion, especially around Lean, that part of the community will think "yes, we know". The new parts come in where Agda is special - constructive, proofs emphasize equational reasoning, universes, etc. What you were expressing was too vague for the current state of knowledge.

Reply:
Okay, but I would still like to express this in the conclusion.

My hope is that some mathematicians (particularly universal algebraists) will be interested in this paper because of the title and the big result we claim to have formalizeead.  But these folks will likely skim over most of the paper understanding very little of the type theory and Agda code.   I expect they will only read the intro and conclusion and try to decide whether it's worth investing their time reading the whole paper in earnest and/or learning type theory, Agda, etc.

Reply:
added conclusion section (needs work)